robottelo/config/base.py

```diff
@@ -931,20 +931,43 @@ def __init__(self, *args, **kwargs):
         self.netmask = None
         self.gateway = None
         self.bridge = None
+        self.network = None
+        self.dhcp_from = None
+        self.dhcp_to = None
+        self.dns_primary = None
 
     def read(self, reader):
         """Read Vlan Network settings."""
         self.subnet = reader.get('vlan_networking', 'subnet')
         self.netmask = reader.get('vlan_networking', 'netmask')
         self.gateway = reader.get('vlan_networking', 'gateway')
         self.bridge = reader.get('vlan_networking', 'bridge')
+        self.network= reader.get('vlan_networking', 'network')
+        self.dhcp_from = reader.get('vlan_networking', 'dhcp_from')
+        self.dhcp_to = reader.get('vlan_networking', 'dhcp_to')
+        self.dns_primary = reader.get('vlan_networking', 'dns_primary')
 
     def validate(self):
         """Validate Vlan Network settings."""
         validation_errors = []
-        if not all(vars(self).values()):
+        if (self.bridge and self.network) or (not self.bridge and not self.network):
+            validation_errors.append(
+                    'exactly one of the "bridge" or "network" parameters '
+                    'must be specified'
+                    )
+        if (self.dhcp_from and not self.dhcp_to) or (not self.dhcp_from and self.dhcp_to):
+            validation_errors.append(
+                    'both or none of "dhcp_from", "dhcp_to" parameters'
+                    'must be specified')
+        params = vars(self).copy()
```

Comment:
```python
ignored = ['bridge', 'network', 'dhcp_from', 'dhcp_to', 'dns_primary']
if not all(value for (key, value) in vars(self).items() if key not in ignored):
```

Maybe you also want to ignore `dns_secondary` if you implement it.

Reply:
yeah, but perhaps let's add it when it gets implemented then